src/main/java/net/coderbot/batchedentityrendering/mixin/MixinLevelRenderer_EntityListSorting.java

```diff
@@ -21,24 +20,25 @@
  */
 @Mixin(value = LevelRenderer.class, priority = 1001)
 public class MixinLevelRenderer_EntityListSorting {
-    @Redirect(method = "renderLevel", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/multiplayer/ClientLevel;entitiesForRendering()Ljava/lang/Iterable;"))
-    private Iterable<Entity> batchedentityrendering$sortEntityList(ClientLevel clientLevel) {
-        // Sort the entity list first in order to allow vanilla's entity batching code to work better.
-        Iterable<Entity> entityIterable = clientLevel.entitiesForRendering();
+	@Shadow
+	private ClientLevel level;
 
-        clientLevel.getProfiler().push("sortEntityList");
+	@ModifyVariable(method = "renderLevel", at = @At(value = "INVOKE_ASSIGN", target = "Ljava/lang/Iterable;iterator()Ljava/util/Iterator;"), ordinal = 0)
```

Comment:
.iterator() is a very common and general method call, so using it as an injection point and only relying on the ordinal is dangerous. I'd recommend using a Slice instead of an ordinal to narrow down the injection points, and then using `allow = 1` on the Redirect to assert that we think that there is only one .iterator() call within our relevant slice. Doing both of those things will make this injection a lot less fragile.

Reply:
Done.